Feature/pipelex get instance

lchoquel opened the pull request:
### 📝 Description

- Make it safer and more tolerant to add domains from toml
- Pipelex.get_instance()

### 🔄 Type of Change
- [ ] 🐛 Bug fix
- [ ] ✨ New feature
- [ ] 💥 Breaking change
- [ ] 📚 Documentation update
- [X] 🧹 Code refactor
- [ ] ⚡ Performance improvement
- [ ] ✅ Test update

lchoquel committed:
Pipelex get_instance

lchoquel committed:
Make it safer and more tolerant to add domains from toml

thomashebrard commented on pipelex/core/domain_library.py:
```diff
@@ -58,6 +60,7 @@ def get_domain(self, domain_code: str) -> Optional[Domain]:
     def get_required_domain(self, domain_code: str) -> Domain:
         the_domain = self.get_domain(domain_code=domain_code)
         if not the_domain:
+            pretty_print(self.root, title="DomainLibrary")
```
The issues is for users that will have tens of domains, it might flood their terminal, or logging file

lchoquel replied:
fixed in subsequent PR

lchoquel committed:
Replace pdf dpi magic numbers by a config default (except in pytests), made it everywhere 72 instead of 300

lchoquel committed:
In CustomBaseModel, truncate fields named base_64 or url when they contain too many bytes.
Made Stuff a CustomBaseModel so that it applies to logging and pretty_printing Stuff.

lchoquel committed:
Control nb of items in dry run lists

lchoquel committed:
Made polyfactory use examples in mock llm object generator.
Defined FORCE_DRY_RUN_MODE_ENV_KEY = "PIPELEX_FORCE_DRY_RUN_MODE".
